src/object.d

```diff
@@ -3190,11 +3199,24 @@ void __ctfeWriteln(T...)(auto ref T values) { __ctfeWrite(values, "\n"); }
  * Create RTInfo for type T
  */
 
+template RTInfoImpl(size_t[] pointers)
+{
+    immutable size_t[pointers.length] data = pointers[];
+    immutable RTInfoImpl = data.ptr;
```

Comment:
> The length is stored in the first word of the pointer bitmap.

Then that's close to what I was suggesting, lest the variable length encoding. I have a little time tomorrow that I can spent on profiling the current implementation. Let's see if that leads to more concrete ideas.

>> Let's assume large objects with GC references are rare for now.

> Arrays (e.g. of strings) can often get quite large. But I'm ok with leaving this optimization for later.

But that's an array of a simple type, not a large object. You can squeeze the pointer bitmap for 16-byte into 2 bits. The threshold for replacing a pointer bitmap with an 8-byte pointer would be a bitmap with 64 bits, i.e. an object with 512 bytes, those are rare.

> That option isn't available on every platform. We could use a separate section with 1 byte alignment, though.

Yes, we should consider a separate section for the metadata at some point, that could also allow a few other optimizations (e.g. 4-byte offsets into that section rather than 8-byte pointers). It'll also ensure that the metadata is packed and not intermingled w/ other data.